### ISSUE TITLE
Support filename hash

### ISSUE THREAD

**Issue by alex996** [jleeson/rollup-plugin-import-css#12]
Is there a way to emit a filename like `custom.xyz123ab.css` instead of `bundle.css`? I tried setting `output: "[name].[hash][extname]"` but got `[name].css` as the result. Also, [`output.assetFileNames`](https://rollupjs.org/configuration-options/#output-assetfilenames) seems to be ignored - perhaps its value could be read by default. Thank you.

**Comment by jleeson** [comment]
Hey @alex996, thanks for opening this issue. I can work on adding this functionality to the plugin when I get time. I will update here when its added.

**Comment by jleeson** [comment]
@alex996, I went ahead and added support for `output.assetFileNames`, you should be able to have your css bundle be named through this option, just make sure you update the plugin to v3.3.0.

Let me know if you are still running into problems.

**Comment by alex996** [comment]
Hi @jleeson, thanks a lot for working on this! Unfortunately, I'm running into another issue with v3.3.0. Using a minimal config

```js
export default {
  input: "src/main.js",
  output: [{ dir: "dist" }], // assetFileNames defaults to "assets/[name]-[hash][extname]"
  plugins: [css()],
};
```

I get the following output:

```sh
dist/
  assets/
    bundle-7510cd7d # missing the file extension
  main.js
```

The issue might be on [this line](https://github.com/jleeson/rollup-plugin-import-css/blob/c9b74ef450030557a28615d8bbe62a3afb54d1d5/src/index.js#L80). Perhaps it should just be:

```diff
const getAssetName = () => {
-  const fileName = options.output ?? (opts.file ?? "bundle.js");
-  return path.basename(fileName, path.extname(fileName));
+  return options.output ?? "bundle.css";
}
```

This is how [`rollup-plugin-scss`](https://github.com/thgh/rollup-plugin-scss/blob/dd41d2d7a4ce924a0f57e7e59cf9c7b0edd9e146/index.ts#L74) does it. Also, I don't believe `opts.file` is a valid config option (was it deprecated?):

```
(!) You have passed an unrecognized option
Unknown input options: file.
```

One more question: I am not well-versed with authoring rollup plugins, but is there a benefit to passing both `name` and `fileName` to [`this.emitFile`](https://rollupjs.org/plugin-development/#this-emitfile)? The docs mention how

> When emitting chunks or assets, either a `name` or a `fileName` can be supplied. If a `fileName` is provided, it will be used unmodified as the name of the generated file

While Rollup doesn't throw when both are passed, `fileName` will always win. IMHO it makes it harder to debug when `getAssetFileName()` sometimes returns undefined, in which case `name` takes over `fileName`. But in other cases `fileName` is not undefined, so it overwrites `name`. Is there a good reason to set both?

Off-topic: please consider not minifying the source code for distribution, it just makes it easier to inspect 😊

Thank you!

**Comment by jleeson** [comment]
Hey @alex996, you are correct in regards to the missing file extension for `getAssetName`. The `opts.file` is not a plugin config option but rather it is the rollup `output.file` option. `opts` is the output config that rollup passes into the plugin hook. So in this case when using the `output.file` option it will generate the css bundle with a similar name to the js bundle.

When it comes to passing both `name` and `fileName` to the `emitFile` function, I do that in order to preserve the functionality of the plugin before the `output.assetFileName` support was added. `fileName` will override everything so  when the `options.assetFileName` is used, it will set fileName to undefined, and the `name` gets set as the bundle name, replacing `[name]` in the `output.assetFileName` option, when `fileName` is used however, `name` gets ignored.

The file extension problem should now be resolved in v3.3.1
